File: src/IsbInterop.CSharp/Accessory/Wrappers/Contents.cs
using IsbInterop.Base;

namespace IsbInterop.Accessory.Wrappers
{
  /// <summary>
  /// Обертка над IContents.
  /// </summary>
  internal class Contents<TI> : ForEach<TI>, IContents<TI> where TI : IObjectInfo
  {
    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="contentsObject">Объект IContents.</param>
    /// <param name="scope">Область видимости.</param>
    public Contents(object contentsObject, IScope scope) : base(contentsObject, scope) { }
  }
}

File: src/IsbInterop/Accessory/Proxies/Contents.vb
Imports IsbInterop.Base

Namespace Accessory.Proxies
    ''' <summary>
    ''' Обертка над IContents.
    ''' </summary>
    Friend Class Contents(Of TI As IObjectInfo)
        Inherits ForEach(Of TI)
        Implements IContents(Of TI)

        ''' <summary>
        ''' Конструктор.
        ''' </summary>
        ''' <param name="contentsObject">Объект IContents.</param>
        ''' <param name="scope">Область видимости.</param>
        Public Sub New(contentsObject As Object, scope As IScope)
            MyBase.New(contentsObject, Scope)
        End Sub
    End Class
End Namespace
